recipe-appTests/listRecipesTest.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  listRestaurantsTest.swift
+//  recipe-appTests
+//
+//  Created by Arthur Hermann on 27/11/2024.
+//
+
+import XCTest
+import Testing
+@testable import recipe_app
+
+final class listRestaurantsTest: XCTestCase {
+    
+    func testRecipeListNotEmpty() {
+        DispatchQueue.global(qos: .background).async {
+            let vm = RecipeListViewModel()
+            vm.searchRecipe("pasta")
+            XCTAssertFalse(vm.recipes.isEmpty)
+            
+        }
+    }
+    
+    func testCorrectRecipeTitle() {
+        DispatchQueue.global(qos: .background).async {
+            let vm = RecipeListViewModel()
+            vm.searchRecipe("pasta")
+            let recipes = vm.recipes
+            for recipe in recipes {
+                XCTAssertTrue(recipe.title.contains("pasta"))
+            }
+        }
+    }
+    func testCorrectNumberOfRecipes() {
+        DispatchQueue.global(qos: .background).async {
+            let vm = RecipeListViewModel()
```

Comment:
The `RecipeListViewModel` keeps its state (like recipes) between tests, which could cause one test’s data to leak into another.

Reset the state before each test. You can reinitialize the view model for each test or add cleanup steps in a setup/teardown method.

https://developer.apple.com/documentation/xctest/xctestcase/set_up_and_tear_down_state_in_your_tests